src/main/java/reposense/parser/CsvParser.java

```diff
@@ -223,17 +226,34 @@ private String getRowContentAsRawString(final CSVRecord record) {
      * Generates map of column header to position number for input {@code possibleHeader}.
      * @throws InvalidCsvException if {@code possibleHeader} does not contain all the mandatory headers.
      */
-    private void validateHeader(String[] possibleHeader) throws InvalidCsvException {
+    private void validateHeader(String[] possibleHeader) throws InvalidCsvException, InvalidHeaderException {
         int headerSize = possibleHeader.length;
+        Set<String> knownColumns = new HashSet<>();
+        StringBuilder unknownColumns = new StringBuilder();
         for (int i = 0; i < headerSize; i++) {
             String possible = possibleHeader[i].trim();
             for (String parsedHeader : mandatoryAndOptionalHeaders()) {
                 if (possible.equalsIgnoreCase(parsedHeader)) {
                     headerMap.put(parsedHeader, i);
+                    knownColumns.add(possible);
                     break;
                 }
             }
+            if (!knownColumns.contains(possible)) {
+                if (unknownColumns.toString().equals("")) {
+                    unknownColumns.append(possible);
+                } else {
+                    unknownColumns.append(", ");
+                    unknownColumns.append(possible);
```

Comment:
Per the above comment, this should be able to be simplified into something that does not need an if-else statement.